diff --git a/dist-cjs/resolveProcessCredentials.js b/dist-cjs/resolveProcessCredentials.js
new file mode 100644
index 0000000000000000000000000000000000000000..75231e07f4b53f5fe931ec8381147c9c230cff8d
--- /dev/null
+++ b/dist-cjs/resolveProcessCredentials.js
@@ -0,0 +1,8 @@
+"use strict";
+Object.defineProperty(exports, "__esModule", { value: true });
+exports.resolveProcessCredentials = void 0;
+const credential_provider_process_1 = require("@aws-sdk/credential-provider-process");
+const resolveProcessCredentials = (options, profile) => {
+    return (0, credential_provider_process_1.fromProcess)(options, profile)();
+};
+exports.resolveProcessCredentials = resolveProcessCredentials;
diff --git a/dist-cjs/resolveProfileData.js b/dist-cjs/resolveProfileData.js
index 4ade404f8ced23eaccc156aa26643b160284da04..1e1c70c1ad0f2c9c02a633cb852df079d2db98ba 100644
--- a/dist-cjs/resolveProfileData.js
+++ b/dist-cjs/resolveProfileData.js
@@ -4,6 +4,7 @@ exports.resolveProfileData = void 0;
 const property_provider_1 = require("@aws-sdk/property-provider");
 const resolveAssumeRoleCredentials_1 = require("./resolveAssumeRoleCredentials");
 const resolveSsoCredentials_1 = require("./resolveSsoCredentials");
+const resolveProcessCredentials_1 = require("./resolveProcessCredentials");
 const resolveStaticCredentials_1 = require("./resolveStaticCredentials");
 const resolveWebIdentityCredentials_1 = require("./resolveWebIdentityCredentials");
 const resolveProfileData = async (profileName, profiles, options, visitedProfiles = {}) => {
@@ -11,6 +12,9 @@ const resolveProfileData = async (profileName, profiles, options, visitedProfile
     if (Object.keys(visitedProfiles).length > 0 && (0, resolveStaticCredentials_1.isStaticCredsProfile)(data)) {
         return (0, resolveStaticCredentials_1.resolveStaticCredentials)(data);
     }
+    if (typeof data.credential_process === 'string') {
+        return (0, resolveProcessCredentials_1.resolveProcessCredentials)(options, profileName);
+    }
     if ((0, resolveAssumeRoleCredentials_1.isAssumeRoleProfile)(data)) {
         return (0, resolveAssumeRoleCredentials_1.resolveAssumeRoleCredentials)(profileName, profiles, options, visitedProfiles);
     }
diff --git a/package.json b/package.json
index e693a9245921f41829a29426d597e7376cc0a4e2..9ccc745617b689046090c97639d08b8779ab71ed 100644
--- a/package.json
+++ b/package.json
@@ -25,6 +25,7 @@
   "dependencies": {
     "@aws-sdk/credential-provider-env": "3.78.0",
     "@aws-sdk/credential-provider-imds": "3.81.0",
+    "@aws-sdk/credential-provider-process": "3.78.0",
     "@aws-sdk/credential-provider-sso": "3.105.0",
     "@aws-sdk/credential-provider-web-identity": "3.80.0",
     "@aws-sdk/property-provider": "3.78.0",
